plugins/extractors/googleworkspace/googleworkspace.go

```diff
@@ -0,0 +1,214 @@
+	var status string
+	e.emit = emit
+	r, err := FetchUsers(ctx, e.TokenSource)
+	if err != nil {
+		return err
+	}
+
+	if len(r.Users) == 0 {
+		e.logger.Info("No users found.\n")
+	} else {
+		for _, u := range r.Users {
+			if !u.Suspended {
+				status = "not suspended"
+			} else {
+				status = "suspended"
+			}
+
+			var userAttributes = make(map[string]interface{})
+			userAttributes = buildOrgAttributes(userAttributes, u.Organizations)
+			userAttributes = buildRelationsAttributes(userAttributes, u.Relations)
+			userAttributes = buildCustomSchemasAttributes(userAttributes, u.CustomSchemas)
+			userAttributes["org_unit_path"] = u.OrgUnitPath
+			userAttributes["aliases"] = strings.Join(u.Aliases, ",")
+
+			e.emit(models.NewRecord(&assetsv1beta1.User{
+				Resource: &commonv1beta1.Resource{
+					Service: "google workspace",
+					Name:    u.Name.FullName,
+					Urn:     u.PrimaryEmail,
+				},
+				Email:    u.PrimaryEmail,
+				FullName: u.Name.FullName,
+				LastName: u.Name.FamilyName,
+				Status:   status,
+				Properties: &facetsv1beta1.Properties{
+					Attributes: utils.TryParseMapToProto(userAttributes),
+				},
+			}))
+		}
+	}
+
+	return nil
+}
+
+// init registers the extractor to catalog
+func init() {
+	if err := registry.Extractors.Register("googleworkspace", func() plugins.Extractor {
+		return New(plugins.GetLog())
+	}); err != nil {
+		panic(err)
+	}
+}
+
+func FetchUsers(ctx context.Context, ts oauth2.TokenSource) (*admin.Users, error) {
+	srv, err := admin.NewService(ctx, option.WithTokenSource(ts))
+	if err != nil {
+		return nil, err
+	}
+
+	r, err := srv.Users.List().Customer("my_customer").OrderBy("email").Do()
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to retrieve users in domain")
+	}
+
+	return r, nil
+}
+
+func buildOrgAttributes(userAttributes map[string]interface{}, i interface{}) map[string]interface{} {
+	if i != nil {
+		itr := reflect.ValueOf(i)
+		if itr.Kind() == reflect.Slice {
+			for idx := 0; idx < itr.Len(); idx++ {
+				valMap := reflect.ValueOf(itr.Index(idx).Interface())
+				for _, key := range valMap.MapKeys() {
+					strct := valMap.MapIndex(key)
+					userAttributes[fmt.Sprintf("%v", key.Interface())] = strct.Interface()
+				}
+			}
+		}
+	}
+	return userAttributes
+}
+
+func buildRelationsAttributes(userAttributes map[string]interface{}, i interface{}) map[string]interface{} {
+	if i != nil {
+		itr := reflect.ValueOf(i)
+		if itr.Kind() == reflect.Slice {
+			for idx := 0; idx < itr.Len(); idx++ {
+				valMap := reflect.ValueOf(itr.Index(idx).Interface())
+				var relationType, relationValue string
+				for _, key := range valMap.MapKeys() {
+					strct := valMap.MapIndex(key)
+					if key.Interface().(string) == "type" {
+						relationType = strct.Interface().(string)
+					} else if key.Interface().(string) == "value" {
+						relationValue = strct.Interface().(string)
+					}
+				}
+				userAttributes[relationType] = relationValue
+			}
+		}
```

Comment:
Some of these iterations are quite nested. can you check if it's possible to refactor them. 
These func's - 

`buildOrgAttributes`
`buildRelationsAttributes`
`buildCustomSchemasAttributes`